src/overlays/actors/ovl_Obj_Switch/z_obj_switch.h

```diff
@@ -17,21 +17,21 @@ typedef enum {
 } ObjSwitchType;
 
 typedef enum {
-    /* 0 */ OBJSWITCH_SUBTYPE_FLOOR_0,
-    /* 1 */ OBJSWITCH_SUBTYPE_FLOOR_1,
-    /* 2 */ OBJSWITCH_SUBTYPE_FLOOR_2,
-    /* 3 */ OBJSWITCH_SUBTYPE_FLOOR_3
+    /* 0 */ OBJSWITCH_SUBTYPE_FLOOR_NORMAL,     // Normal Yellow Switch
+    /* 1 */ OBJSWITCH_SUBTYPE_FLOOR_TOGGLE,     // On/Off Red Switch
+    /* 2 */ OBJSWITCH_SUBTYPE_FLOOR_RESET_OFF,  // Reset when not stood on, blue switch (default state: off)
+    /* 3 */ OBJSWITCH_SUBTYPE_FLOOR_RESET_ON    // Reset when not stood on, blue switch (default state: on)
 } ObjSwitchSubTypeFloor;
 
 typedef enum {
-    /* 0 */ OBJSWITCH_SUBTYPE_EYE_0,
-    /* 1 */ OBJSWITCH_SUBTYPE_EYE_1
+    /* 0 */ OBJSWITCH_SUBTYPE_EYE_NORMAL,       // Normal Yellow Eye
+    /* 1 */ OBJSWITCH_SUBTYPE_EYE_TOGGLE        // On/Off Grey Eye
 } ObjSwitchSubTypeEye;
 
 typedef enum {
-    /* 0 */ OBJSWITCH_SUBTYPE_CRYSTAL_0,
-    /* 1 */ OBJSWITCH_SUBTYPE_CRYSTAL_1,
-    /* 4 */ OBJSWITCH_SUBTYPE_CRYSTAL_4 = 4
+    /* 0 */ OBJSWITCH_SUBTYPE_CRYSTAL_NORMAL,   // Normal
+    /* 1 */ OBJSWITCH_SUBTYPE_CRYSTAL_TOGGLE,   // On/Off Blue/Red Crystal
+    /* 4 */ OBJSWITCH_SUBTYPE_CRYSTAL_TIED = 4  // Tied to Switch Flag's State
```

Comment:
this subtype's behavior is pretty weird and I think it should be explained more
for example, what does hitting such a switch even do

Reply:
- hitting this subtype of switch will act as normal
- if you hit another switch that shares the same switch flag (only the flag matters) it will update with the switch flag (so flag set = crystal on, flag unset = crystal off)

However toggle switches won't be updated with the switch flag, if you trigger a switch that's not updated it will do nothing at all, not even a cutscene, the state will update though

One use case I see is using it with the toggle subtype, which can prevent the player from using the tied subtype but you could use it later, can be used with a gate for example

Reply:
Looking at the spawn data, there are only two instances of subtype 4 being used in vanilla and master quest. 

In vanilla, there's a room in the water temple where you have to jump into a pool with a fast moving current, equip irons, and shoot a crystal switch that's underwater inside a dragon's mouth to open up a timed gate to reach a key. A second crystal switch on the opposite side of the gate lets you open the gate again and swim back. 

In master quest, there is a room in spirit temple with two crystal switches that opens up a gate meant to drop a like like on your head. The second crystal allows you to open the gate if you've entered from above.

So the general purpose is to sync up the state of two crystal switches, rather than have them act independently.